src/Simplex_tree/include/gudhi/Simplex_tree/Simplex_tree_iterators.h

```diff
@@ -85,6 +85,11 @@ class Simplex_tree_boundary_simplex_iterator : public boost::iterator_facade<
   typedef typename SimplexTree::Vertex_handle Vertex_handle;
   typedef typename SimplexTree::Siblings Siblings;
 
+  Simplex_tree_boundary_simplex_iterator()
```

Comment:
I assume cython is forcing you to do that? I'd like to see a comment next to it saying that it is added just for cython, and that the object it initializes should be overwritten ASAP and never used before it is overwritten.

Reply:
Comment added on 1f31e5ce as cython is trying to do:
```cpp
new((void*)&(p->__pyx_v_it)) Gudhi::Simplex_tree_interface<Gudhi::Simplex_tree_options_full_featured>::Boundary_simplex_iterator();
```